### ISSUE TITLE
Restructuring and refactoring the bes-inference package.

### ISSUE THREAD

**Issue by asztalosors** [plasma-sds/bes-inference#12]
The bes-inference package's current version is outdated and requires restructuring and refactoring. The main guideline by which the refactoring should occur. The package is based on Tensorflow. This is a follow-up for the now obsolete #10.

![Image](https://github.com/user-attachments/assets/7af0f424-8df0-488a-897a-308bbc8c8030)

Detailed structure:

1.  networks - is a subpackage aimed at containing the various NN that are featured in the development of the package. These are the following: Extreem Learning Machines (ELM) #5 , Multi Layer Perceptron networks (MLP) #13  and convolutional NN (CNN)
2. inference - subpackage that should hold the code to build the network to be used for inference, that may be either for density inference or emission inference #15 . The subpackage holds a universal data object to be used in BES inference #14 . For both emission and/or density inference capable network generation, the subpackage needs to process the data object, select the network architecture, and train and set the NN.
3. functions - is a subpackage that features data loading and sorting routines, evaluation functions to process and compare NN #7 and physics-informed loss functions #6 .

**Comment by asztalosors** [comment]
The above-described concept is obsolete. The code will be restructured in issue: #16
